modules/programs/docker-cli.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    mkOption
+    ;
+
+  cfg = config.programs.docker-cli;
+
+  jsonFormat = pkgs.formats.json { };
+in
+{
+  meta.maintainers = [ lib.maintainers.friedrichaltheide ];
+
+  options.programs.docker-cli = {
+    settings = mkOption {
+      type = jsonFormat.type;
+      default = { };
+      example = ''
+        {
+          "proxies" = {
+            "default" = {
+              "httpProxy" = "http://proxy.example.org:3128";
+              "httpsProxy" = "http://proxy.example.org:3128";
+              "noProxy" = "localhost";
+            };
+          };
+      '';
+      description = ''
+        Available configuration options for the Docker CLI see:
+        <https://docs.docker.com/reference/cli/docker/#docker-cli-configuration-file-configjson-properties
+      '';
+    };
+  };
+
+  config = {
+    home.file = lib.optionalAttrs (cfg.settings != { }) {
+      ".docker/config.json" = {
```

Comment:
Looking at the referenced upstream documentation, it would be nice to have an option for choosing the directory location to populate `DOCKER_CONFIG` and create the file at that location